File: client/moonloader/galileo/PedState.lua
local PedState = {}
PedState.__index = PedState

function PedState.new(state, isInInterior, vehicleType)
    local object = setmetatable({}, PedState)
    object.state = state
    object.isInInterior = isInInterior
    object.vehicleType = vehicleType

    return object
end

function PedState.parse(table)
    local state = table.state
    local isInInterior = table.isInInterior
    local vehicleType = table.vehicleType

    return PedState.new(state, isInInterior, vehicleType)
end

function PedState:__tostring()
    return  "state="..tostring(self.state)..
            ", isInInterior="..tostring(self.isInInterior)..
            ", vehicleType="..tostring(self.vehicleType)
end

return PedState

File: server/src/main/java/galileo/server/model/PedState.java
package galileo.server.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PedState {
    @JsonProperty("state")
    private String state;
    @JsonProperty("vehicleType")
    private String vehicleType;
    @JsonProperty("isInInterior")
    private boolean isInInterior;
}
